File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/java/domain/hierarchy/AbstractNodeAttribute.java
package com.coupang.springframework.data.requery.java.domain.hierarchy;

import com.coupang.kotlinx.objectx.ToStringBuilder;
import com.coupang.springframework.data.requery.domain.AbstractPersistable;
import io.requery.*;
import lombok.Getter;
import lombok.Setter;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

/**
 * com.coupang.springframework.data.requery.java.domain.hierarchy.AbstractNodeAttribute
 *
 * @author debop
 * @since 18. 6. 4
 */
@Entity
public class AbstractNodeAttribute extends AbstractPersistable<Long> {

    private static final long serialVersionUID = 7424000305448372730L;

    @Getter
    @Key
    @Generated
    @Column(name = "attr_id")
    protected Long id;

    @Column(name = "attr_name")
    protected String name;

    @Column(name = "attr_value")
    protected String value;

    @ManyToOne
    @ForeignKey
    protected AbstractTreeNode node;

    @Override
    public int hashCode() {
        return Objects.hash(name);
    }

    @Transient
    @Override
    protected @NotNull ToStringBuilder buildStringHelper() {
        return super.buildStringHelper()
            .add("name", name)
            .add("value", value);
    }
}

File: spring-data-requery-kotlin/src/test/kotlin/com/coupang/springframework/data/requery/domain/hierarchy/AbstractNodeAttribute.kt
package com.coupang.springframework.data.requery.domain.hierarchy

import com.coupang.kotlinx.objectx.ToStringBuilder
import com.coupang.springframework.data.requery.domain.AbstractPersistable
import io.requery.*
import java.util.*

@Entity
abstract class AbstractNodeAttribute: AbstractPersistable<Long>() {

    @get:Key
    @get:Generated
    @get:Column(name = "attr_id")
    abstract val id: Long?


    @get:Column(name = "attr_name")
    abstract var name: String?

    @get:Column(name = "attr_value")
    abstract var value: String?


    @get:ManyToOne
    @get:ForeignKey
    abstract var node: AbstractTreeNode?

    override fun hashCode(): Int {
        return Objects.hash(name)
    }

    @Transient
    override fun buildStringHelper(): ToStringBuilder {
        return super.buildStringHelper()
            .add("name", name)
            .add("value", value)
    }
}